src/communication/channelmanager.hpp

```diff
@@ -51,23 +68,60 @@ class ChannelManager : public ChannelManagerItf {
      * @param transport communication transport.
      * @return * aos::Error
      */
-    aos::Error Init(TransportItf& transport);
+    Error Init(TransportItf& transport);
 
     /**
      * Create channel with dedicated port.
      *
      * @param port port to bind channel.
-     * @return aos::RetWithError<ChannelItf&>
+     * @return aos::RetWithError<ChannelItf*>
      */
-    aos::RetWithError<ChannelItf&> CreateChannel(uint32_t port) override;
+    RetWithError<ChannelItf*> CreateChannel(uint32_t port) override;
 
     /**
      * Deletes channel.
      *
      * @param port port channel is bound to.
      * @return aos::Error
      */
-    aos::Error DeleteChannel(uint32_t port) override;
+    Error DeleteChannel(uint32_t port) override;
+
+    /**
+     * Connects to communication channel.
+     *
+     * @return aos::Error.
+     */
+    Error Connect() override;
```

Comment:
it looks like connect/close method could be private

Reply:
this method will be used in client side